src/plugins/Workers/TransactionSyncStreamWorker/methods/startHistoricalSync.js

```diff
@@ -60,6 +60,10 @@ module.exports = async function startHistoricalSync(network) {
     });
   }
 
+  const header = await this.transport.getBlockHeaderByHeight(bestBlockHeight);
```

Comment:
Let's move it to `getBestBlockHeightFromTransport` call to reduce possibility of race conditions (reorgs).

Reply:
No. It's still the same problem we always have. We need to implement getBlockHashByHeight. 

The way we have the bestBlockHeight in param is already by calling getBlockHeaderByHeight, what we need here is to set the hash from this specific height.

Reply:
Well, because that would be un-necessary if the syncup failed, and as the height stays constants during the whole historical sync, setting the hash/height would mean next time you won't historically sync from where you should but instead from hash/height set.
 
So in all case, set need to be done at the end. 
In all case, height stays constant, so no race-conditions. 

Do you want me to do the header call on top ? 98% of the case it will just works, but it happens quite some time that historical sync fail.

Reply:
I see your point. Agree!